docs: Add docs for GenSX Cloud (#589)
gensx-inc/gensx: website/docs/src/content/cloud/projects-environments.mdx

---
title: Projects and environments
description: Organize your GenSX workflows and deployments
---

# Projects and environments

> GenSX organizes your workflows and deployments using a flexible structure of projects and environments, making it easy to match the rest of your application architecture and CI/CD topology.

Reviewer: It might be worth adding some sort of sentence that quickly explains that "projects" are the top level resource and "environments" are underneath them; might not be super intuitive to everyone so it's a good expectation to set up front.

## Project structure

A project in GenSX is a collection of related workflows that are deployed, managed, and monitored together:

- **Projects as logical units**: Group related workflows that serve a common purpose
- **Shared configuration**: Apply settings across all workflows in a project
- **Collective deployment**: Deploy all workflows within a project in one operation
- **Unified monitoring**: View traces and metrics for an entire project

Projects typically correspond to a codebase or application that contains multiple workflows.

## Environment separation

> Within each project, you can have multiple environments:
> 
> - **Development**: For building and testing new features
> - **Staging**: For pre-production validation
> - **Production**: For live, user-facing workflows
> - **Custom environments**: Create any specialized environments you need

Reviewer: Same feedback as on another comment. This makes me think that gensx automatically creates these three envs for me. 

```suggestion
Within each project, you can have multiple environments. For example, you could create three environments for each project:

- **Development**: For building and testing new features
- **Staging**: For pre-production validation
- **Production**: For live, user-facing workflows

You have full control over your environments so you can organize them however you see fit. 
```

- Workflow deployments
- Configuration and environment variables
- Execution traces and monitoring data

## Configuring projects

### Project configuration file

Projects are defined using a `gensx.yaml` file at the root of your codebase:

```yaml
# gensx.yaml
projectName: customer-support-bot
description: AI assistant for customer support
```

This configuration applies to both local development and cloud deployments.

## Working with environments

### Deploying to different environments

Deploy your workflows to specific environments using the CLI:

```bash
# Deploy to the default environment
gensx deploy src/workflows.tsx

# Deploy to a staging environment
gensx deploy src/workflows.tsx --environment staging

# Deploy to production
gensx deploy src/workflows.tsx --environment production
```

### Environment-specific configuration

Set environment-specific variables during deployment:

```bash
# Development-specific settings
gensx deploy src/workflows.tsx --environment development \
  --env LOG_LEVEL=debug \
  --env OPENAI_MODEL=gpt-4

# Production-specific settings
gensx deploy src/workflows.tsx --environment production \
  --env LOG_LEVEL=error \

>   --env OPENAI_MODEL=gpt-4-turbo

Reviewer: Maybe this should be `OPENAI_API_KEY` instead? if we leave it as model, we should make them `gpt-4o` or `gpt-4.1`

```

## Projects in the GenSX Console

The GenSX Console organizes everything by project and environment:

> ![Project structure in console](/cloud/project-structure.png)

Reviewer: this screenshot is unclear to me. Maybe we need red arrows that point out project name and env name on the screenshot. It's especially confusing because we don't discuss "default" env anywhere so readers might not make the connection that "default" is the environment name

Author: Yeah, we are missing the ability to create new environments from the CLI so I couldn't construct more meaningful data. Once we ship that PR, we need to come back and retake this screenshot.

Selecting an environment brings you to the workflows view:

> ![Environment page](/cloud/console-workflows.png)

Reviewer: i think these screenshots are backwards

> From here you can trigger a workflow within the console for workflows deployed to GenSX Cloud serverless:

Reviewer: ```suggestion
When you click into a workflow, you can trigger it within the console if you've deployed it to GenSX Cloud:
```

![Triggering a workflow in the console](/cloud/console-playground.png)

> You can also see API documentation and sample code for triggering those workflows:

Reviewer: ```suggestion
You can also see API documentation and sample code for calling that workflow:
```
